pkg/mutation/assignmetadata.go

```diff
@@ -0,0 +1,72 @@
+package mutation
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MetadataMutator is a mutator wrapping the AssignMetadata type.
+type AssignMetadataMutator struct {
+	//assignMetadata *mutationsv1.AssignMetadata
+	Path  parser.Path
+	Value string
+}
+
+func (m AssignMetadataMutator) Matches(obj metav1.Object, ns *corev1.Namespace) bool {
+	// TODO: add match logic
+	return false
+}
+
+func (m AssignMetadataMutator) Mutate(obj *unstructured.Unstructured) error {
+	return mutate(obj.Object, m.Path.Nodes, m.Value)
+}
+
+func mutate(current interface{}, remainingPaths []parser.Node, value string) error {
+	pathEntry := remainingPaths[0]
+	if len(remainingPaths) == 1 {
+		// TODO: handle types other than object
```

Comment:
We should flesh this out the mutation TODOs before committing.

Reply:
Deleted comment. For now we can only have object as the last element of the path.